scripts/runtime_mgmt/update-runtime.py

```diff
@@ -0,0 +1,104 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import argparse
+from pathlib import Path
+
+import requests
+from azure.ai.ml import MLClient, load_environment
+from azure.identity import AzureCliCredential
+
+ENVIRONMENT_YAML = Path(__file__).parent / "runtime-env" / "env.yaml"
+
+EXAMPLE_RUNTIME_NAME = "example-runtime-ci"
+TEST_RUNTIME_NAME = "test-runtime-ci"
+
+
+class PFSRuntimeHelper:
+    def __init__(self, ml_client: MLClient):
+        subscription_id = ml_client._operation_scope.subscription_id
+        resource_group_name = ml_client._operation_scope.resource_group_name
+        workspace_name = ml_client._operation_scope.workspace_name
+        location = ml_client.workspaces.get(name=workspace_name).location
+        self._request_url_prefix = (
+            f"https://{location}.api.azureml.ms/flow/api/subscriptions/{subscription_id}"
+            f"/resourceGroups/{resource_group_name}/providers/Microsoft.MachineLearningServices"
+            f"/workspaces/{workspace_name}/FlowRuntimes"
+        )
+        token = ml_client._credential.get_token("https://management.azure.com/.default").token
+        self._headers = {"Authorization": f"Bearer {token}"}
+
+    def update_runtime(self, name: str, env_asset_id: str) -> None:
+        body = {
+            "runtimeDescription": "Runtime hosted on compute instance, serves for examples checks.",
+            "environment": env_asset_id,
+            "instanceCount": "",
+        }
+        response = requests.put(
+            f"{self._request_url_prefix}/{name}",
+            headers=self._headers,
+            json=body,
+        )
+        response.raise_for_status()
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--path", help="Path to config.json", type=str)
+    return parser.parse_args()
+
+
+def init_ml_client(
+    subscription_id: str,
+    resource_group_name: str,
+    workspace_name: str,
+) -> MLClient:
+    return MLClient(
+        credential=AzureCliCredential(),
+        subscription_id=subscription_id,
+        resource_group_name=resource_group_name,
+        workspace_name=workspace_name,
+    )
+
+
+def create_environment(ml_client: MLClient) -> str:
+    environment = load_environment(source=ENVIRONMENT_YAML)
+    env = ml_client.environments.create_or_update(environment)
+    # get workspace id from REST workspace object
+    resource_group_name = ml_client._operation_scope.resource_group_name
+    workspace_name = ml_client._operation_scope.workspace_name
+    location = ml_client.workspaces.get(name=workspace_name).location
```

Comment:
small knowledge: you can avoid setting the name parameter here.

Reply:
learnt that, interesting